internal/session/provider_config.go

```diff
@@ -64,13 +74,79 @@ func NewProviderConfig(configuration schema.SessionConfiguration) ProviderConfig
 		}
 		config.EncodeFunc = serializer.Encode
 		config.DecodeFunc = serializer.Decode
-	} else { // if no option is provided, use the memory provider.
+	case len(configuration.Memcache) > 0: // If memcache configuration is provided, then use the memcache provider.
+		providerName = "memcache"
+		serializer := NewEncryptingSerializer(configuration.Secret)
```

Comment:
the encode and decode function can be factorize. Why not move the instantiation of serializer and move
```
config.EncodeFunc = serializer.Encode
config.DecodeFunc = serializer.Decode
```
out of the switch block?